### ISSUE TITLE
Cannot import accessors at the main module level

### ISSUE THREAD

**Issue by salman451** [wavespectra/wavespectra#15]
Hi,
I have been having installation issues with pip install wavespectra==3.5.3. numcodecs==0.6.4 fails.

So, I am separately first installing the following modified dependencies, which install without a problem:

attrdict
click
cmocean
dask>=0.17.1
gcsfs
fsspec
matplotlib
numpy>=1.14
pandas
python-dateutil
pyyaml
sortedcontainers
scipy
toolz
xarray<0.15.1
numcodecs==0.6.3
zarr==2.3.2

Following this I install wavespectra as pip install wavespectra. Again no issues.

When I start python interpretor: import wavespectra results in the following:

.....lib/python3.6/site-packages/wavespectra/__init__.py:13: UserWarning: Cannot import accessors at the main module level
  warnings.warn("Cannot import accessors at the main module level")
Cannot import reading function read_triaxys:
cannot import name '_easy_facetgrid'
Cannot import reading function read_dataset:
cannot import name '_easy_facetgrid'
Cannot import reading function read_spotter:
cannot import name '_easy_facetgrid'
Cannot import reading function read_ww3_msl:
cannot import name '_easy_facetgrid'
Cannot import reading function read_era5:
cannot import name '_easy_facetgrid'
Cannot import reading function read_netcdf:
cannot import name '_easy_facetgrid'
Cannot import reading function read_ncswan:
cannot import name '_easy_facetgrid'
Cannot import reading function read_wwm:
cannot import name '_easy_facetgrid'
Cannot import reading function read_swan:
cannot import name '_easy_facetgrid'
Cannot import reading function read_wavespectra:
cannot import name '_easy_facetgrid'
Cannot import reading function read_dictionary:
cannot import name '_easy_facetgrid'
Cannot import reading function read_ww3:
cannot import name '_easy_facetgrid'

of course, nothing really works from this point onwards.

Would greatly appreciate a solution.

Best,
Salman.

**Comment by rafa-guedes** [comment]
Hi Salman,

thanks for sending this issue. This sounds like an issue with xarray version, this `_easy_facetgrid` function is imported from xarray. I realise now we have fixed the version of xarray to be smaller than `0.15.1` (because this version has some breaking changes we have not fixed yet) but we should probably fix the minimum version as well. We will make sure these versions will be more carefully established. For now, please try updating `xarray==0.15.0` to see if that will fix your problem.

On the other issue you mentioned with `numcodecs==0.6.4`, I can't seem to be able to reproduce it on my system. I also have this version of numcodecs and wavespectra installs with no issue (on ubuntu20.04). What OS are you trying to install it on, and which python version do you have please?

Cheers,
Rafael

**Comment by salman451** [comment]
Hi Rafa,

Thanks for this feedback - very useful. In fact, I have been using facetgrid plotting capability that you wrapped for wavespectra - it is very useful, but I wasn't able to guess if it were xarray version problem.

I have got it working with the following default.txt modifications:
attrdict
click
cmocean
dask>=0.17.1
gcsfs
fsspec
**matplotlib==2.2.2**
**numpy>=1.14**
**pandas>=0.25**
python-dateutil
pyyaml
sortedcontainers
scipy
toolz
**xarray==0.15.0**
**numcodecs==0.6.3**
**zarr==2.3.2**

My main issues here were:
(i) numcodecs==0.6.4 doesn't install and is needed for zarr, so I restricted zarr==2.3.2 to prev version that works with numcodec==0.6.3, which installs without a problem.
(ii) xarray problems as mentioned above
(iii) matplotlib==2.2.2, I was running an older version which gave an error when using facetgrid
(iv) numpy and pandas I upgraded as I read some dependency errors during installations.

It would be great if the versions of the packages could also be included as there can be compatibly issues. We inherit most of our python packages from module environment - many of which are older versions. Without version information, the requirements are satisfied because of existing older packages, only for the code to break later. But, may be it is quite difficult for you to tailor it that way.

My python version is 3.6.1. I think the python interpreter in our module env was compiled on intel processors, and that seems to be problematic with numcodecs==0.6.4. Not sure.

As the above is working - I am going to close the issue and thank you for your swift reply.

Cheers,
Salman.
